lib/identity/client.js

```diff
@@ -67,6 +68,60 @@ class Client extends PartialClient {
     options.headers.Authorization = `Bearer ${token}`
     return fetch(url, options)
   }
+
+  async createSecret(secret) {
+    const trimmedName = secret.secretName.trim()
+    const trimmedValue = secret.secretValue.trim()
+    if (trimmedName === '') {
+      throw { message: 'Name cannot be empty' }
+    }
+    const re = new RegExp('^[a-zA-Z0-9-_]{1,30}$')
+    if (!re.test(trimmedName)) {
+      throw {
+        message: 'Secret name must contain alphanumeric characters, -, or _',
+      }
+    }
+    if (trimmedValue === '') {
+      throw { message: 'Value cannot be empty' }
+    }
+    const groupName = [`tozny.secret.${this.storage.config.clientId}`]
+    // check if the clients secret group already exists
+    let groups = await this.storage.listGroups(
+      this.storage.config.clientId,
+      groupName
+    )
+    let group
+    let groupMembers = []
+    if (groups.groups.length < 1) {
+      // create the group if it doesn't exist
+      let newGroup = await this.storage.createGroup(groupName[0])
+      group = newGroup.group
+      let capabilities = {
+        read: true,
+        share: true,
+      }
+      var member = new GroupMember(this.storage.config.clientId, capabilities)
+      groupMembers.push(member)
+      // add read and share capabilities for the client
+      await this.storage.addGroupMembers(group.groupID, groupMembers)
+    } else {
+      group = groups.groups[0]
+    }
+    // create the record
+    const recordType = `tozny.secret.ae41d9cc-cd5c-4a90-8efe-0046784e195c.${secret.secretType}.${secret.secretName}`
```

Comment:
make this value `ae41d9cc-cd5c-4a90-8efe-0046784e195c` a constant so it can be reused by other secret methods (such as list or describe secret)

Reply:
Is this a UUIDv5? Where ever it is defined, I would like a code comment or some form of documentation explaining how it works (why this UUID).

Reply:
this was just a randomly generated uuid, there's no special significance to this one; the goal was just to make sure that a client won't accidentally make another group with the exact same name. 
i can definitely add a comment about that if that's helpful.

Reply:
it can't be just "a" randomly generated uuid. It needs to be THE random uuid that is used for secrets across the js-sdk or any other SDK. Similar pattern that we used for note signatures

https://github.com/tozny/internal-docs/blob/ae0831ffc8bb45cca228114157dba86ba2a3faa7/tozny-platform/notes/tozny-field-signing.md#tozny-field-signing-protocol-with-blake2b-and-ed25519

Reply:
okay, good to know. i will update that.

Reply:
@lkwdwrd do we need it to be a UUIDv5, can it just be a UUIDv4 that we know is constant across all of our SDKs that use secrets? At the end of the day we need to generate a UUIDv4 for the UUIDv5 so I am not sure what the additional value is. But this mostly just a curiosity of my own.

Reply:
Down the line if we have a whole bunch of these namespaces, we can ship a single UUID namespace and calculate the UUIDv5 for each type within that namespace. It can be a little cleaner, and more meaningful, but it doesn't have to be. We already have a tozny UUID namespace from TFSV1, so it wouldn't be too hard to re-use that and just calculate the v5 for some string.

Ultimately, it doesn't matter one bit as long as they are well documented constants.

I don't feel strongly that it needs to be a v5, its just an option.

Reply:
So you are thinking pseudocode wise TSV1 does
`namespace = "794253a4-310b-449d-9d8d-4575e8923f40"
sigtag = uuid.v5("TFSP1;ED25519;BLAKE2B", namespace)`

so toznysecrets should do
`namespace = "794253a4-310b-449d-9d8d-4575e8923f40"
sigtag = uuid.v5("tozny.secrets", namespace)`

I think that makes good sense.
Any reason an online tool like https://www.uuidtools.com/v5 can't be used to generate it? it doesn't seem the standard golang implementation supports it.

@lkwdwrd

Reply:
exactly -- and yes that online generator is just fine to create it. We can even hard code the UUID for the time being with a code comment to indicate where it came from. Calculation becomes more valuable as we have more of them, but I have a sneaking suspicion as we go forward... we will have more of these.